gnucash/import-export/aqb/gnc-flicker-gui.c

```diff
@@ -0,0 +1,518 @@
+                                         (GSourceFunc) time_handler,
+                                         (gpointer) widget);
+}
+
+/* Initialize the drawingarea to black and paint the flickerchallenge */
+static gboolean
+on_flicker_challenge_draw(GtkWidget *widget, cairo_t *cr,
+                          __attribute__((unused)) gpointer user_data)
+{
+    do_flicker_drawing(widget, cr);
+
+    return FALSE;
+}
+
+/* called when the drawing area is destroyed */
+static void
+on_flicker_challenge_destroy(GtkWidget *widget,
+                             __attribute__((unused)) gpointer user_data)
+{
+    /* remove the timeout function */
+    g_source_remove(flickerdraw.interval);
+}
+
+/* Initialize the drawing area "flicker marker" in black and draw the marker for
+ * the position of the TAN-Generator */
+static gboolean
+on_flicker_marker_draw(__attribute__((unused)) GtkWidget *widget, cairo_t *cr,
+                       __attribute__((unused)) gpointer data)
+{
+    do_marker_drawing(cr);
+
+    return FALSE;
+}
+
+/* This signal is emitted when the drawing area "flicker marker" is visible */
+static void
+on_flicker_marker_map(GtkWidget *widget)
+{
+    /* Set the height of the drawing area */
+    gtk_widget_set_size_request(widget, -1, flickerdraw.y_barpos);
+}
+
+/* The value for "Field width" has been changed on the spin button and the
+ * flicker display is updated */
+static void
+on_spin_barwidth_value_changed(GtkSpinButton *spin, GtkWidget *widget)
+{
+    flickerdraw.barwidth = gtk_spin_button_get_value_as_int(spin);
+    flickerdraw.x_drawpos = (flickerdraw.width - 4 * flickerdraw.margin -
+                             5 * flickerdraw.barwidth) / 2;
+
+    /* Moving the position triangles */
+    gtk_widget_queue_draw(widget);
+}
+
+/* The Button-Press-Event of the Spinbutton "Barwidth" */
+static gboolean
+on_spin_barwidth_button_press_event(__attribute__((unused)) GtkSpinButton *spin,
+                                    GdkEventButton *event,
+                                    __attribute__((unused)) gpointer user_data)
+{
+    /* disable the use of the third mouse button */
+    if (event->button == 3)
+    {
+        return TRUE;
+    }
+    return FALSE;
+}
+
+/* The value for "waiting time" was changed on the spin button and
+ * the speed of the flickering display is updated */
+static void
+on_spin_delay_value_changed(GtkSpinButton *spin, GtkWidget *widget)
+{
+    flickerdraw.delay = gtk_spin_button_get_value_as_int(spin);
+
+    flickerdraw.change_interval = TRUE;
+    time_handler(widget);
+}
+
+/* The Button-Press-Event for the Spinbutton "Delay" */
+static gboolean
+on_spin_delay_button_press_event(__attribute__((unused)) GtkSpinButton *spin,
+                                 GdkEventButton *event,
+                                 __attribute__((unused)) gpointer user_data)
+{
+    /* disable the use of the third mouse button */
+    if (event->button == 3)
```

Comment:
Please use `GDK_BUTTON_SECONDARY` instead of the magic number. Your comment should explain why you want to disable the go to max/min operation. Both apply to `on_spin_barwidth_button_press_event`

Reply:
I will optimize the min / max settings to reasonable values and remove the functions for blocking the second button.